src/main/java/ee/tuleva/onboarding/payment/provider/montonio/MontonioOrderCreator.java

```diff
@@ -0,0 +1,127 @@
+      return String.format("member:%s", paymentData.getRecipientPersonalCode());
+    }
+
+    return String.format(
+        "30101119828, IK:%s, EE3600001707", paymentData.getRecipientPersonalCode());
+  }
+
+  @SneakyThrows
+  public MontonioOrder getOrder(PaymentData paymentData, Person person) {
+    MontonioPaymentChannel paymentChannelConfiguration =
+        montonioPaymentChannelConfiguration.getPaymentProviderChannel(
+            paymentData.getPaymentChannel());
+
+    BigDecimal amount = getPaymentAmount(paymentData);
+    Currency currency = paymentData.getCurrency();
+
+    return MontonioOrder.builder()
+        .accessKey(paymentChannelConfiguration.getAccessKey())
+        .merchantReference(paymentInternalReferenceService.getPaymentReference(person, paymentData))
+        .returnUrl(getPaymentSuccessReturnUrl(paymentData.getType()))
+        .notificationUrl(getNotificationUrl())
+        .grandTotal(amount)
+        .currency(currency)
+        .exp(clock.instant().getEpochSecond() + 600)
+        .locale(getLanguage())
+        .payment(
+            MontonioPaymentMethod.builder()
+                .amount(amount)
+                .currency(currency)
+                .methodOptions(
+                    MontonioPaymentMethodOptions.builder()
+                        .preferredProvider(paymentChannelConfiguration.getBic())
+                        .preferredLocale(getLanguage())
+                        .paymentDescription(getEpisPaymentDescription(paymentData))
+                        .build())
+                .build())
+        .build();
+  }
+
+  private String getPaymentSuccessReturnUrl(PaymentType paymentType) {
+    if (paymentType == PaymentType.MEMBER_FEE) {
+      return apiUrl + "/payments/member-success";
+    }
+
+    return apiUrl + "/payments/success";
+  }
+
+  private BigDecimal getPaymentAmount(PaymentData paymentData) {
+    if (paymentData.getType() == PaymentData.PaymentType.MEMBER_FEE) {
+      return this.getMemberPaymentAmount(paymentData);
+    }
+
+    if (paymentData.getAmount() == null) {
+      throw new IllegalArgumentException("Payment amount must not be null");
+    }
+
+    return paymentData.getAmount();
+  }
+
+  private BigDecimal getMemberPaymentAmount(PaymentData paymentData) {
+    if (memberFee == null) {
+      throw new IllegalArgumentException("Member fee must not be null");
+    }
+    if (Objects.equals(paymentData.getRecipientPersonalCode(), memberFeeTestPersonalCode)
+        && memberFeeTestPersonalCode != null) {
+      return BigDecimal.ONE;
+    }
+    return memberFee;
+  }
+
+  private String getLanguage() {
+    Locale locale = localeService.getCurrentLocale();
+    return Locale.ENGLISH.getLanguage().equals(locale.getLanguage()) ? "en" : locale.getLanguage();
+  }
+
+  private String getNotificationUrl() {
+    if (useFakeNotificationsUrl) {
+      // Montonio doesn't support localhost notification urls
+      return "https://tuleva.ee/fake-notification-url";
```

Comment:
Can we parameterize the whole `notificationUrl` instead of the boolean flag `useFakeNotificationsUrl`?